microscopium/_util.py

```diff
@@ -1,6 +1,82 @@
 import numpy as np
 import numbers
 
+
+def generate_spiral(dim, direction, clockwise=False):
+    """Generate an dim * dim of sequental integers in a spiral pattern.
+
+    Many HCS systems use a spiral pattern to capture multiple fields in
+    for each well. This function generates a spiral pattern of integers
+    to be used by the montaging functions.
+
+    Parameters
+    ----------
+    dim : int
+        The dimension of the spiral. The function will return a dim * dim
+        array.
```

Comment:
Or just select the starting co-ordinate based on the direction chosen? Let's say I want to generate the BBBC017 stitching pattern  ``[[2, 3, 4], [1, 0, 5]]``

If I call ``generate(2, 3, "left", True)``, it's safe to assume I want the BBBC stitching pattern so it should start at [1, 0].

Come to think of it, inferring the starting co-ordinate form the directions means it'll work for shapes that aren't odd too!! eg ``generate(2, 2, "up", False)`` should generate a "quadrant" stitch pattern.. ``[[2, 1], [3, 0]]`` :D